lightning/src/ln/channel.rs

```diff
@@ -4020,6 +4013,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 	}
 }
 
+fn is_unsupported_shutdown_script(their_features: &InitFeatures, scriptpubkey: &Script, is_outbound: bool) -> bool {
+	// BOLT 2 says we must only send a scriptpubkey of certain standard forms,
+	// which for a a BIP-141-compliant witness program is at max 42 bytes in length.
+	// So don't let the remote peer feed us some super fee-heavy script.
+	let is_script_too_long = is_outbound && scriptpubkey.len() > 42;
```

Comment:
`is_outbound` isn't relevant here - see first half of https://github.com/rust-bitcoin/rust-lightning/pull/794#discussion_r576876359

Reply:
Hmmmm, I don't understand. I thought you meant that the check had to be moved into the function?

Reply:
That's what the second part suggested. I didn't fully understand what you really were after with the first part of the comment :\

Reply:
Tried to clarify on slack, sorry for the confusion. My point there was that I think we should do the script_too_long check in both the outbound and inbound channels, so no need for an is_outbound check here. This is a change against master, but not in practice, since the list of standard output types all come with a strict length check.